linera-views/src/key_value_store_view.rs

```diff
@@ -668,6 +668,52 @@ where
         Ok(self.context.contains_key(&key).await?)
     }
 
+    /// Tests whether the view contains a range of indices
+    /// ```rust
+    /// # tokio_test::block_on(async {
+    /// # use linera_views::memory::create_memory_context;
+    /// # use linera_views::key_value_store_view::KeyValueStoreView;
+    /// # use crate::linera_views::views::View;
+    /// # let context = create_memory_context();
+    ///   let mut view = KeyValueStoreView::load(context).await.unwrap();
+    ///   view.insert(vec![0,1], vec![42]).await.unwrap();
+    ///   let keys = vec![vec![0,1], vec![0,2]];
+    ///   let results = view.contains_keys(keys).await.unwrap();
+    ///   assert_eq!(results, vec![true, false]);
+    /// # })
+    /// ```
+    pub async fn contains_keys(&self, indices: Vec<Vec<u8>>) -> Result<Vec<bool>, ViewError> {
+        let mut results = Vec::with_capacity(indices.len());
+        let mut missed_indices = Vec::new();
+        let mut vector_query = Vec::new();
+        for (i, index) in indices.into_iter().enumerate() {
+            if index.len() > self.max_key_size() {
```

Comment:
We use `ensure!` for checks like this in many other places. Maybe it's more intuitive, because it shows the guarantee, not the error case.

Reply:
Ok.